### PR TITLE
Promotion from AMD internal branch for 2022.Q3.5

### PR DESCRIPTION
**Pull request by WenqingLiAMD**
@s-perron

### CODE REVIEW

**Comment by s-perron** [comment]
I'm on leave until January.  Please ask @vettoreldaniele or @dnovillo for reviews.

File: include/vkgcDefs.h [hunk 1]
```diff
@@ -750,6 +752,15 @@ struct PipelineShaderOptions {
 
   /// Override value for ThreadGroupSizeZ
   unsigned overrideShaderThreadGroupSizeZ;
+
+  ///< When there is a valid "feedback loop" in renderpass, lateZ needs to be enabled
```
Comment (Flakebi) [inline include/vkgcDefs.h:756]: The leading `<` on the comment should be removed.

File: llpc/context/llpcPipelineContext.cpp [hunk 2]
```diff
@@ -143,18 +143,6 @@ static cl::opt<bool> ScalarizeWaterfallDescriptorLoads("scalarize-waterfall-desc
...
-    "TraceRayInlineAmdInternal",  // RT_ENTRY_TRACE_RAY_INLINE,
-    "",                           // RT_ENTRY_TRACE_RAY_HIT_TOKEN,
-    "RayQueryProceedAmdInternal", // RT_ENTRY_RAY_QUERY_PROCEED,
-    "",                           // RT_ENTRY_INSTANCE_INDEX,
-    "",                           // RT_ENTRY_INSTANCE_ID,
-    "",                           // RT_ENTRY_OBJECT_TO_WORLD_TRANSFORM,
-    "",                           // RT_ENTRY_WORLD_TO_OBJECT_TRANSFORM,
-};
-#endif
 #endif
```
Comment (Flakebi) [inline llpc/context/llpcPipelineContext.cpp:146]: The empty `#if VKI_RAY_TRACING` can be removed.

**Comment by amdvlk-admin** [comment]
### Test summary for commit fa495412509e6ffb60208b9bda71764a2be7c155
<details><summary>CTS tests (Failed: 0/285657)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.6, Gfx8</summary><ul>

<li>Passed: 36222/72463 (50.0%)</li>
<li>Failed: 0/72463 (0.0%)</li>
<li>Not Supported: 36241/72463 (50.0%)</li><li>Warnings: 0/72463 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi21</summary><ul>

<li>Passed: 40419/72473 (55.8%)</li>
<li>Failed: 0/72473 (0.0%)</li>
<li>Not Supported: 32054/72473 (44.2%)</li><li>Warnings: 0/72473 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 37135/72477 (51.2%)</li>
<li>Failed: 0/72477 (0.0%)</li>
<li>Not Supported: 35342/72477 (48.8%)</li><li>Warnings: 0/72477 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 34834/68244 (51.0%)</li>
<li>Failed: 0/68244 (0.0%)</li>
<li>Not Supported: 33410/68244 (49.0%)</li><li>Warnings: 0/68244 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by vettoreldaniele** [comment]
LGTM

**Comment by amdvlk-admin** [comment]
### Test summary for commit 205e80785cef09a0cdbeec088af1f4b547258de0
<details><summary>CTS tests (Failed: 0/285657)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.6, Gfx8</summary><ul>

<li>Passed: 36222/72463 (50.0%)</li>
<li>Failed: 0/72463 (0.0%)</li>
<li>Not Supported: 36241/72463 (50.0%)</li><li>Warnings: 0/72463 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi21</summary><ul>

<li>Passed: 40419/72473 (55.8%)</li>
<li>Failed: 0/72473 (0.0%)</li>
<li>Not Supported: 32054/72473 (44.2%)</li><li>Warnings: 0/72473 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 37135/72477 (51.2%)</li>
<li>Failed: 0/72477 (0.0%)</li>
<li>Not Supported: 35342/72477 (48.8%)</li><li>Warnings: 0/72477 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 34834/68244 (51.0%)</li>
<li>Failed: 0/68244 (0.0%)</li>
<li>Not Supported: 33410/68244 (49.0%)</li><li>Warnings: 0/68244 (0.0%)</li>
</ul></details>


</ul></details>

**Review by Flakebi — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]

**Pull request closed without merging**